### PR TITLE
Thread pool

### PR DESCRIPTION
**Pull request by wirew0rm**
Ported from opencmw including porting of tests from catch2 to ut
Renamed to fit into the project conventions and enabled emscripten
compatibility.

TODO:
- [x] emscripten support

### CODE REVIEW

**Comment by wirew0rm** [comment]
Added the test proposed in https://github.com/fair-acc/opencmw-cpp/issues/243 to the threadpool.

File: test/qa_thread_affinity.cpp [hunk 1]
```diff
@@ -0,0 +1,161 @@
+#include <boost/ut.hpp>
```
Comment (RalphSteinhagen) [inline test/qa_thread_affinity.cpp:1]: the qa_thread_affinity unit-test should be added to the CI/CMakeList.txt.

**Review by RalphSteinhagen — APPROVED** [review]
OK, thanks for migrating this from opencmw-cpp and making it compatible with WASM! :+1:

The qa_thread_affinity unit-test should be added to the CI/CMakeList.txt. 

Otherwise OK!

File: test/qa_thread_affinity.cpp (hunk 1 above)
Comment (wirew0rm) [reply test/qa_thread_affinity.cpp:1]: thanks for catching that, should be fixed.

**Pull request merged**